libs/store_helper/user/ebpf_store_helper.cpp

```diff
@@ -0,0 +1,5 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// Include the same C code as CPP
+#include "..\ebpf_store_helper.c"
```

Comment:
Why do we need this wrapper cpp file?

Reply:
By just including the .c file, in UM the compiler would otherwise build c-decl and the linking on projects such e.g. `netebpfext_user.lib` would not resolve externals.

Reply:
That sounds like a bug, most likely due to not putting
extern "C" {} in ebpf_store_helper.h like we do in most other .h files.

Reply:
Not a bug, as `extern "c"{}` was initially tried:

```c
#ifdef USER_MODE
#ifdef __cplusplus
extern "C"
{
#endif
#endif
...
#ifdef USER_MODE
#ifdef __cplusplus
}
#endif
#endif
```

but would cause the following mirror-type of errors on projects linking the library, e.g.:

```
8>------ Rebuild All started: Project: netebpfext_unit, Configuration: Debug x64 ------
8>ebpf_store_helper_um.lib(ebpf_store_helper.obj) : error LNK2019: unresolved external symbol close_registry_key referenced in function ebpf_store_open_or_create_provider_registry_key
8>ebpf_store_helper_um.lib(ebpf_store_helper.obj) : error LNK2019: unresolved external symbol write_registry_value_binary referenced in function ebpf_store_update_helper_prototype
```

Thus, I believe the cleanest (working) solution (and less strain for the eye in `ebpf_store_helper.h`) is a simple cpp wrapper.

Reply:
> 8>ebpf_store_helper_um.lib(ebpf_store_helper.obj) : error LNK2019: unresolved external symbol close_registry_key referenced in function ebpf_store_open_or_create_provider_registry_key

Sounds like a bug.  You correctly have `extern "C"` in ebpf_registry_helper_km.h but it's missing from ebpf_registry_helper_um.h, which I think is the bug causing that unresolved symbol.

I disagree that a cpp wrapper is clean.  It's inconsistent with the rest of the project, and also results in name mangling that makes some debugging harder, without any real benefit.

Reply:
Actually, `extern "c"` should not bein `ebpf_registry_helper_km.h ` (as it's always C), it's a leftover of many tests, thanks for pointing it out, now removed.

The cpp file is now removed.